src/main/java/com/conveyal/gtfs/validator/PatternFinderValidator.java

```diff
@@ -47,47 +60,73 @@ public void complete(ValidationResult validationResult) {
         // TODO should we really return them in the validation result, or just put them in the DB?
         // The thing is, in a relational database this would require at least two tables and extra columns in the trips.
         // patterns, stopsInPatterns, patternForTrip. Though we could in fact get the stop sequence from any example trip in the pattern.
+        // FIXME: There may be a better way to handle getting the full list of stops
+        Map<String, Stop> stopById = new HashMap<>();
+        for (Stop stop : feed.stops) {
+            stopById.put(stop.stop_id, stop);
+        }
         // FIXME In the editor we need patterns to exist separately from and before trips themselves, so me make another table.
-        List<Pattern> patterns = patternFinder.createPatternObjects();
+        List<Pattern> patterns = patternFinder.createPatternObjects(stopById, errorStorage);
+        Connection connection = null;
         try {
             // TODO this assumes gtfs-lib is using an SQL database and not a MapDB.
             // Maybe we should just create patterns in a separate step, but that would mean iterating over the stop_times twice.
             LOG.info("Storing pattern ID for each trip.");
-            Connection connection = feed.getConnection();
+            connection = feed.getConnection();
             Statement statement = connection.createStatement();
             String tripsTableName = feed.tablePrefix + "trips";
+            // FIXME: use Table class to define and populate patterns and pattern_stops tables?
             String patternsTableName = feed.tablePrefix + "patterns";
             String patternStopsTableName = feed.tablePrefix + "pattern_stops";
+//            String patternGeometryTableName = feed.tablePrefix + "pattern_geometry";
             statement.execute(String.format("alter table %s add column pattern_id varchar", tripsTableName));
-            statement.execute(String.format(
-                    "create table %s (pattern_id varchar primary key, route_id varchar, description varchar)",
-                    patternsTableName));
-            statement.execute(String.format(
-                    "create table %s (pattern_id varchar, stop_sequence integer, stop_id varchar, " +
-                            "primary key (pattern_id, stop_sequence))",
-                    patternStopsTableName));
+            // FIXME: Here we're creating a pattern table that has an integer ID field (similar to the other GTFS tables)
+            // and a varchar pattern_id with essentially the same value cast to a string. Perhaps the pattern ID should
+            // be a UUID or something, just to better distinguish it from the int ID?
+            statement.execute(String.format("create table %s (id serial, pattern_id varchar primary key, " +
```

Comment:
I don't think we necessarily need it.  I was unsure whether to keep it for backwards compatibility with the previous Pattern structure.  It may be helpful to keep just because every other table has a similar definition (int id and key field varchar entity_id).  Without a key field on the pattern table, we would need to rethink the Table class and some of its methods (e.g., `getKeyField`).

Reply:
Let's discuss

Reply:
Keeping this field for now so that references between Entities for the editor are consistently strings and not the ID int field.